source/qb64pe.bas

```diff
@@ -26063,6 +26077,40 @@ SUB increaseUDTArrays
     REDIM _PRESERVE udtenext(x + 1000) AS LONG
 END SUB
 
+FUNCTION CompareVersions (v1$, v2$)
+    DIM AS LONG p1, p2, p3, p4, p5, p6 'part 1 through 6
+
+    IF GetVersionValues(v1$, p1, p2, p3) = 0 THEN CompareVersions = 666: EXIT FUNCTION 'Evil Error code.  Something went wrong!
+    IF GetVersionValues(v2$, p4, p5, p6) = 0 THEN CompareVersions = 666: EXIT FUNCTION 'Evil Error code.  Something went wrong!
+
+    IF p1 < p4 THEN CompareVersions = -1: EXIT FUNCTION
+    IF p1 > p4 THEN CompareVersions = 1: EXIT FUNCTION
+    IF p2 < p5 THEN CompareVersions = -1: EXIT FUNCTION
+    IF p2 > p5 THEN CompareVersions = 1: EXIT FUNCTION
+    IF p3 < p6 THEN CompareVersions = -1: EXIT FUNCTION
+    IF p3 > p6 THEN CompareVersions = 1: EXIT FUNCTION
+
+    CompareVersions = 0 'if all else fails, the two versions are equal.
+END FUNCTION
+
+FUNCTION GetVersionValues (version$, v1 AS LONG, v2 AS LONG, v3 AS LONG)
```

Comment:
I think version comparisons with only two or one digit should be allowed, there's certain kinds of checks that can't easily be done otherwise. Ex. If you want to check for a minor version less than `3.4`, you can't just do `$IF VERSION < 3.4 THEN` so you'd have to basically use an arbitrarily large patch version to try and catch all the potential versions we release, such as `$IF VERSION < 3.3.999 THEN`.